google-cloud-apis/java-apis/maven-apis/src/com/google/cloud/tools/intellij/cloudapis/CloudApiUiPresenter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.cloudapis;
+
+import javax.swing.Icon;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Presenter for core Cloud API UI, allows accessing UI points for libraries for extension points to
+ * update, including documentation links and version info.
+ */
+// TODO: move to core cloud API once dependency is inverted.
+public interface CloudApiUiPresenter {
```

Comment:
I wonder if there is any value in making this more generic - i.e. have the extensions just provide custom UI panels? I'm thinking about the future if we support custom languages which might need more custom tailored UI.

Reply:
also are these really the only two things we need to extract? what about source, api reference etc.?

Reply:
So next (hopefully last) PR will add custom UI panels for Cloud API selector panel, you're correct - so Maven module will "inject" BOM selection. As for source and API reference, see methods of CloudUiPresenter - it allows adding more links for a selected cloud library.

Reply:
you mean `CloudApiUiPresenter`? Probably missing something but I see two methods: `addCloudLibraryDocumentationLink`, and `updateCloudLibraryVersionLabel`

Reply:
Correct, this is the one which allows adding more documentation links to library details panel - `addCloudLibraryDocumentationLink`

Reply:
ah ok. its the name that confused me then. for one, the ui has a link called 'documentation' so thats what I thought it was for. How about just addCloudLibraryLinks?

Reply:
Sounds good, I can also probably make this method accept collection of Optionals like previous code did.